src/types.lisp

```diff
@@ -845,24 +845,38 @@ The foreign array must be freed with foreign-array-free."
 (defmacro with-foreign-slots ((vars ptr type) &body body)
   "Create local symbol macros for each var in VARS to reference
 foreign slots in PTR of TYPE. Similar to WITH-SLOTS.
-Each var can be of the form: 
-  name                       name bound to slot of same name              
-  (:pointer name)            name bound to pointer to slot of same name
-  (name slot-name)           name bound to slot-name
-  (name :pointer slot-name)  name bound to pointer to slot-name"
+Each var can be of the form:
+
+SLOT - SLOT is bound to the slot of the same
+name. i.e. to (FOREIGN-SLOT-VALUE SLOT)
+
+(:pointer SLOT) - SLOT is bound to the pointer to the slot of the same
+name. i.e. to (FOREIGN-SLOT-POINTER SLOT)
+
+(NAME SLOT) - NAME is bound to (FOREIGN-SLOT-VALUE SLOT)
+
+(:POINTER (NAME SLOT)) - NAME is bound to (FOREIGN-SLOT-POINTER SLOT)
```

Comment:
Maybe we can drop this one?

Reply:
I.e. support only `(name (:pointer slot))`.

Reply:
we support both (:pointer slot-name) as well as (slot-name :pointer) 
by means of the :pointer extension. Therefore it seemed like a good idea at that
time to include all these other forms for symmetry and completeness

I'm not sure if we want to support only (slot-name :poiinter) which would be only be
consistent.

Perhaps the docstring could refer to a subset of the supported forms to keep it simple
and the manual could explain the rationale and the remaining forms.

Reply:
I think supporting both was a mistake and neither is documented, so perhaps we can get away with removing `(slot-name :pointer)`.

Reply:
I think it was in the manual - and to be  probably more consistent it would be better to remove (:pointer slot-name) - and retain (slot-name :pointer) as the LHS is the variable, wouldn't it?